### ISSUE TITLE
Fix `unit-no-unknown` false positives for the second and subsequent `image-set()` with `x` descriptor

### ISSUE THREAD

**Issue by chalkygames123** [stylelint/stylelint#6721]
### What steps are needed to reproduce the bug?

Just lint the following CSS with `unit-no-unknown` turned on.

`test.css`:

```css
a {
	background-image: url('first.png'),
		image-set(url('second.png') 1x, url('second@2x.png') 2x);
	background-image: url('first.png'), url('second.png'),
		image-set(url('third.png') 1x, url('third@2x.png') 2x);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "unit-no-unknown": [
      true
    ]
  }
}
```

### How did you run Stylelint?

CLI with `stylelint test.css`

### Which version of Stylelint are you using?

`15.3.0`

### What did you expect to happen?

No problems to be reported.

### What actually happened?

The `unit-no-unknown` rule reports the following problems when using the `x` unit in a value for an `image-set()` function.

```
test.css
 3:32  ✖  Unexpected unknown unit "x"  unit-no-unknown
 3:57  ✖  Unexpected unknown unit "x"  unit-no-unknown
 5:31  ✖  Unexpected unknown unit "x"  unit-no-unknown
 5:55  ✖  Unexpected unknown unit "x"  unit-no-unknown
```

Related: https://github.com/stylelint/stylelint/pull/4654

### Does the bug relate to non-standard syntax?

No.

**Comment by ybiquitous** [comment]
@chalkygames123 Thanks for the report using the template. We can reproduce the problem with the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgIyTAGsBzAJwgFdMoBaASwFskSZ4VKyAbACgHIAZvTIBnHADoADphJ8AlABosuXExYxaImDh6defLZBpSZ8lAEYAHgo7d+hiDQACAJksnZclG7kBuZfiEpBTUdGqs7Hr8QqIS0p42UQYwRlAe8krYOKrMrJraunZ8OAAWwmnxZlaJRaXlru6VXj7+mAC+IAogQlwwAHJIjHCIMJaDkr3iYCIineCOQiQIIBiYKCjoIGSUvSKb7Kvr65vU9Di0mBC01ESXAO7YCCg42zBY621YHV1GiwBiEDIzBwywAViJHHNYJJZohDhsQGIAJ69Lj0TDZJ6bcwAVnEAGZxAAGTZfLAgNpAA).

As a workaround, you can avoid the problem using the [`ignoreFunctions` secondary option](https://stylelint.io/user-guide/rules/unit-no-unknown#ignorefunctions-regex-regex-string) like this:

```json
{
  "rules": {
    "unit-no-unknown": [true, {"ignoreFunctions": ["image-set"]}]
  }
}
```

I guess we can fix the false positive by changing the following code:

https://github.com/stylelint/stylelint/blob/964b1ba71ac3526a1b3fb305e72c3af3fa3c9e24/lib/rules/unit-no-unknown/index.js#L121

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by chalkygames123** [comment]
Thank you for the guidance, I'm on it!

**Comment by chalkygames123** [comment]
@ybiquitous Could you give me some explanation especially about what these lines do because I'm not familiar with the Stylelint codebase?

https://github.com/stylelint/stylelint/blob/964b1ba71ac3526a1b3fb305e72c3af3fa3c9e24/lib/rules/unit-no-unknown/index.js#L133-L135

**Comment by ybiquitous** [comment]
@chalkygames123 The lines you pointed out came from PR #4427. I'm not sure since the PR is old, but maybe I guess the lines handle the following case (see `left 20x` after `image-set()`)

https://github.com/stylelint/stylelint/blob/964b1ba71ac3526a1b3fb305e72c3af3fa3c9e24/lib/rules/unit-no-unknown/__tests__/index.js#L459-L467

This logic's implementation has a few deeper nested `if` blocks, so it seems hard to read. It may be good refactoring timing if you wish.

**Comment by chalkygames123** [comment]
Thank you. So I'll open a draft PR with minimal changes, then let me have some time to refactor it.

**Comment by romainmenke** [comment]
_Was looking at this because it is relevant for the migration to a different parser for media queries._

Why isn't `x` present in the list of known units?
`x` is just an alias for `dppx`.

https://github.com/stylelint/stylelint/blob/main/lib/reference/units.js#L79-L82

```js
	// Resolution
	'dpi',
	'dpcm',
	'dppx',
```

specification : https://drafts.csswg.org/css-values-4/#resolution

mdn : https://developer.mozilla.org/en-US/docs/Web/CSS/resolution

**Comment by ybiquitous** [comment]
> `x` is just an alias for `dppx`.

Oh, I didn't know it. The reason to ignore `x` is to avoid false positives for `image-set()`. See:

- #3424
- #3770

**Comment by romainmenke** [comment]
If I read those correctly the main motivation seems to be that `x` is often a typo for `px` and that `x` is only valid in `image-set()` property values and in the `resolution` media query?

**Comment by ybiquitous** [comment]
@romainmenke Yes, your understanding is correct. 👍🏼
